spds/constraints.md

```diff
@@ -4,4 +4,71 @@ title: Constraints
+An interval constraint is defined by a time offset and a time period in which the policy is switched to an inactive state.
+For the same example, the picture below defines a time offset of 2 time units and a time period of 5 time units.
+During the offset period, the policy is active and adjustments are made normally. 
+After offset ends, the inactivity period starts that lasts for the specified _intervalDuration_.
+
+After the inactivity period ends and the simulation continues, the policy is switched to an active state again and adjustments are made normally.
+However, if the _repeat_ flag is set to true, the interval constraint is applied again with the _offset_ and _intervalDuration_ (shown with a dashed line in the figure).
+
+![example-interval.png](..%2Fimages%2Fconstraints%2Fexample-interval.png)
+
+### Thrashing Constraint
+
+Another temporal constraints that is applicable to the whole [TargetGroup](../docu/#targetsTargetGroup) and restricts the behavior of all the policies applying to it. 
+The main objective behind the thrashing constraint is to avoid oscillations in the system. 
+The thrashing constraint is defined by a minimum time period between two consecutive adjustments with opposite directions.
+The motivation behind defining a thrashing constraint can be the stability of the system and the billing type of the cloud provider.
+Often when resources are provisioned it does not make any sense to deprovision them after a short period of time while you are billed for the next 10 minutes. 
+Hence, the modeller can specify a Thrashing constraint.
+
+The picture below shows the behavior of a policy without a thrashing constraint in which a consecutive scale in is occurring after a scale out.
+![constraint-thrashing.png](..%2Fimages%2Fconstraints%2Fconstraint-thrashing.png)
+
+When adding a Thrashing constraint, the modeler can avoid such an oscillation. The picture below shows the behavior of the same policy with a thrashing constraint of 5 time units.
+![constraint-thrashing-2.png](..%2Fimages%2Fconstraints%2Fconstraint-thrashing-2.png)
+
+
+## State Based Constraints
+
+State based constraints are constraints in which the restriction and behavior alternation considers the model state during the simulation.
+At the moment there is only one state based constraint available, the [TargetGroup Size Constraint](../docu/#constraintstargetTargetGroupSizeConstraint).
+
+### TargetGroup Size Constraint
+
+The TargetGroup Size Constraint is a constraint that is applicable to the whole [TargetGroup](../docu/#targetsTargetGroup) and restricts the behavior of all the policies applying to it.
+The constraint is defined by a minimum and maximum number of elements that the target group can have at any given time. 
+All adjustments to a number of elements outside of the defined range are changed to either scaling to the minimum or the maximum.
+
+![group-size-constraint.png](..%2Fimages%2Fconstraints%2Fgroup-size-constraint.png)
```

Comment:
i really like this min/max representation. 
but i still find details to complain about: the one directional arrow look like a time line at first glance, which is odd. i would make it bidirectional arrows, such that the reader has a visual heads up that yells: "stop, this is probably not time! read the axis description carefully!".